<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DataSourceDialog</class>
 <widget class="QMainWindow" name="DataSourceDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>S-DIA - 데이터 소스</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true">QWidget#centralwidget {
    background-color: #e9ecef;
}
QLabel {
    color: #000000;
    font-size: 14px;
    background-color: transparent;
}
QLineEdit, QTextEdit {
    padding: 8px 12px;
    border: 1px solid #ced4da;
    border-radius: 4px;
    background-color: #ffffff;
    font-size: 14px;
    min-height: 20px;
    color: #495057;
}
QLineEdit:focus, QTextEdit:focus {
    border: 1px solid #80bdff;
    border-color: #80bdff;
    outline: 0;
}
QPushButton {
    color: #495057;
    background-color: #ffffff;
    border: 1px solid #ced4da;
    padding: 8px;
    border-radius: 4px;
    font-size: 14px;
}
QPushButton:hover {
    background-color: #f8f9fa;
}
QPushButton#save_button {
    color: #fff;
    background-color: #80bdff;
    border-color: #80bdff;
    font-weight: bold;
}
QPushButton#save_button:hover {
    color: #fff;
    background-color: #80bdff;
    border-color: #80bdff;
}
QPushButton#save_button:pressed {
    background-color: #80bdff;
    border-color: #80bdff;
}
QFrame#form_container {
    background-color: white;
    border-radius: 8px;
    border: 1px solid #dee2e6;
}</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>10</number>
    </property>
    <property name="leftMargin">
     <number>10</number>
    </property>
    <property name="topMargin">
     <number>10</number>
    </property>
    <property name="rightMargin">
     <number>10</number>
    </property>
    <property name="bottomMargin">
     <number>10</number>
    </property>
    <item>
     <widget class="QWidget" name="title_container" native="true">
      <property name="styleSheet">
       <string notr="true">background-color: transparent;</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="spacing">
        <number>4</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="title_label">
         <property name="styleSheet">
          <string notr="true">font-size: 20px;
font-weight: bold;
color: #000000;
background-color: transparent;</string>
         </property>
         <property name="text">
          <string>데이터 소스 추가</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="form_container">
      <property name="frameShape">
       <enum>QFrame::Shape::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="spacing">
        <number>4</number>
       </property>
       <property name="leftMargin">
        <number>24</number>
       </property>
       <property name="topMargin">
        <number>24</number>
       </property>
       <property name="rightMargin">
        <number>24</number>
       </property>
       <property name="bottomMargin">
        <number>24</number>
       </property>
       <item>
        <widget class="QWidget" name="name_container" native="true">
         <layout class="QHBoxLayout" name="name_layout">
          <property name="spacing">
           <number>2</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QLabel" name="name_text">
            <property name="styleSheet">
             <string notr="true">font-size: 16px;
font-weight: bold;</string>
            </property>
            <property name="text">
             <string>데이터 소스 명</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="name_required">
            <property name="styleSheet">
             <string notr="true">color: #dc3545;
font-size: 16px;
font-weight: bold;</string>
            </property>
            <property name="text">
             <string>*</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>10</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLineEdit" name="name_input">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>600</width>
              <height>38</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>600</width>
              <height>38</height>
             </size>
            </property>
            <property name="placeholderText">
             <string>데이터 소스 명을 입력하세요</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="data_path_container" native="true">
         <layout class="QHBoxLayout" name="desc_layout">
          <property name="spacing">
           <number>2</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QLabel" name="desc_text">
            <property name="styleSheet">
             <string notr="true">font-size: 16px;
font-weight: bold;</string>
            </property>
            <property name="text">
             <string>원본 데이타</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label">
            <property name="styleSheet">
             <string notr="true">color: #dc3545;
font-size: 16px;
font-weight: bold;</string>
            </property>
            <property name="text">
             <string>*</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_5">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>32</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLineEdit" name="data_path_input">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>519</width>
              <height>38</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>557</width>
              <height>38</height>
             </size>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="fs_explorer">
            <property name="minimumSize">
             <size>
              <width>79</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>79</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>찾아보기</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="csv_1_container" native="true">
         <layout class="QHBoxLayout" name="type_layout">
          <property name="spacing">
           <number>2</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QLabel" name="type_text">
            <property name="styleSheet">
             <string notr="true">font-size: 16px;
font-weight: bold;</string>
            </property>
            <property name="text">
             <string>CSV 옵션</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Policy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>58</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>원본데이터</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_6">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Policy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>8</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QComboBox" name="charset_input">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>42</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>70</width>
              <height>38</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">                QComboBox {
                    background-color: white;
                    border: 1px solid #dee2e6;
                    border-radius: 4px;
                    padding: 5px 10px;
                    min-height: 30px;
                    color: #495057;
                }
                QComboBox:focus {
                    border: 1px solid #339af0;
                }
                QComboBox::drop-down {
                    subcontrol-origin: padding;
                    subcontrol-position: center right;
                    width: 30px;
                    border: none;
                }
                QComboBox::down-arrow {
                    image: url(src/ui/resources/images/down-arrow-svgrepo-com.svg);
                    width: 10px;
                    height: 10px;
                    margin-right: 1px;
                }
                QComboBox QAbstractItemView {
                    background-color: white;
                    border: 1px solid #dee2e6;
                    border-radius: 4px;
                    outline: none;
                    selection-background-color: #e9ecef;
                    selection-color: #000;
                }
                QComboBox QAbstractItemView::item {
                    padding: 8px 20px;
                    color: #495057;
                }
                QComboBox QAbstractItemView::item:selected {
                    background-color: #e9ecef;
                    color: #000;
                }</string>
            </property>
            <property name="currentIndex">
             <number>0</number>
            </property>
            <property name="maxVisibleItems">
             <number>13</number>
            </property>
            <property name="insertPolicy">
             <enum>QComboBox::InsertPolicy::NoInsert</enum>
            </property>
            <item>
             <property name="text">
              <string>UTF-8</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>EUC-KR</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer name="type_spacer">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="csv_2_container" native="true">
         <layout class="QHBoxLayout" name="type_layout">
          <property name="spacing">
           <number>2</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>4</number>
          </property>
          <item>
           <spacer name="horizontalSpacer_7">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Policy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>126</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>구분자</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_8">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Policy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>38</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QComboBox" name="separator_input">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>42</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>70</width>
              <height>38</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">                QComboBox {
                    background-color: white;
                    border: 1px solid #dee2e6;
                    border-radius: 4px;
                    padding: 5px 10px;
                    min-height: 30px;
                    color: #495057;
                }
                QComboBox:focus {
                    border: 1px solid #339af0;
                }
                QComboBox::drop-down {
                    subcontrol-origin: padding;
                    subcontrol-position: center right;
                    width: 30px;
                    border: none;
                }
                QComboBox::down-arrow {
                    image: url(src/ui/resources/images/down-arrow-svgrepo-com.svg);
                    width: 10px;
                    height: 10px;
                    margin-right: 1px;
                }
                QComboBox QAbstractItemView {
                    background-color: white;
                    border: 1px solid #dee2e6;
                    border-radius: 4px;
                    outline: none;
                    selection-background-color: #e9ecef;
                    selection-color: #000;
                }
                QComboBox QAbstractItemView::item {
                    padding: 8px 20px;
                    color: #495057;
                }
                QComboBox QAbstractItemView::item:selected {
                    background-color: #e9ecef;
                    color: #000;
                }</string>
            </property>
            <property name="editable">
             <bool>false</bool>
            </property>
            <property name="insertPolicy">
             <enum>QComboBox::InsertPolicy::NoInsert</enum>
            </property>
            <item>
             <property name="text">
              <string>,</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>TAB</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>|</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer name="type_spacer_2">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="csv_3_container" native="true">
         <layout class="QHBoxLayout" name="type_layout">
          <property name="spacing">
           <number>2</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>4</number>
          </property>
          <item>
           <spacer name="horizontalSpacer_10">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Policy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>126</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>헤더여부</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_9">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Policy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>24</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QComboBox" name="has_header_input">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>42</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>70</width>
              <height>38</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">                QComboBox {
                    background-color: white;
                    border: 1px solid #dee2e6;
                    border-radius: 4px;
                    padding: 5px 10px;
                    min-height: 30px;
                    color: #495057;
                }
                QComboBox:focus {
                    border: 1px solid #339af0;
                }
                QComboBox::drop-down {
                    subcontrol-origin: padding;
                    subcontrol-position: center right;
                    width: 30px;
                    border: none;
                }
                QComboBox::down-arrow {
                    image: url(src/ui/resources/images/down-arrow-svgrepo-com.svg);
                    width: 10px;
                    height: 10px;
                    margin-right: 1px;
                }
                QComboBox QAbstractItemView {
                    background-color: white;
                    border: 1px solid #dee2e6;
                    border-radius: 4px;
                    outline: none;
                    selection-background-color: #e9ecef;
                    selection-color: #000;
                }
                QComboBox QAbstractItemView::item {
                    padding: 8px 20px;
                    color: #495057;
                }
                QComboBox QAbstractItemView::item:selected {
                    background-color: #e9ecef;
                    color: #000;
                }</string>
            </property>
            <property name="insertPolicy">
             <enum>QComboBox::InsertPolicy::NoInsert</enum>
            </property>
            <item>
             <property name="text">
              <string>있음</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>없음</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer name="type_spacer_3">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <spacer name="horizontalSpacer_11">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="save_button">
        <property name="minimumSize">
         <size>
          <width>80</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>저장</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="cancel_button">
        <property name="minimumSize">
         <size>
          <width>80</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>취소</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
